### PR TITLE
C51-377: Case List Independent Scrolling + Misc Fixes

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR, Independent Scrolling has been implemented to the Case List when viewing a case

## Before
before

## After
after

## Technical Details
1. A new partial `case-list-table-first-column-header.html` is created, which contains the column header markup for case list's first column. The markup is moved from `case-list-table.directive.html`.
Along with using it in its previous place, it is also used when the Viewing the case details. 
It was necessary to use this marku twice, because otherwise it was not possible to affix the header to the top when scrolling.

2. Also the `civicase__pager` has changed its location, and has been moved outside of `civicase__case-list-table-container`. Reason is similar to the above point,  it was not possible to affix the footer to the bottom when scrolling.

3. In `case-list-table.directive.js`, new logic has been added. When viewing the case details and not expanded, the height of the case list is set to `viewport height - the height headers and footers`
2019-03-11 at 12 42 PM

### Fix 1
There was a flaw of logic in Activity Feed down arrow visibility. When having 26/51/76(25*n + 1) records, the down arrow was not visible when 25*n-1 records were visible, Means the user could never see the last record.

**Before**
2019-03-11 at 12 47 PM

**After**
2019-03-11 at 12 46 PM

**Technical Details**
Logic has been fixed in `activity-feed.directive.js`.

### Fix 2
When Scrolling, the additional tags were not scrolling along with the container.

**Before**
before t

**After**
after t

**Technical Details**
1. In `tags-container.directive.html`, instead of `popover`, the `uib-popover` directive is used, which makes the popup scroll along with its parent.

### CODE REVIEW

File: ang/test/civicase/case/list/directives/sticky-table-header.directive.spec.js [hunk 1]
```diff
@@ -104,27 +104,55 @@ describe('civicaseStickyTableHeader directive', function () {
...
+        scope.viewingCase = false;
+        scope.$digest();
+        $timeout.flush(); // Flushing any timeouts used.
+      }));
 
-    it('adds min-width style to the table elements', function () {
-      expect(element.find('thead').html()).toContain('style="min-width');
-    });
+      it('adds min-width style to the table elements', function () {
+        expect(element.find('thead').html()).toContain('style="min-width');
```
Comment (reneolivo) [inline ang/test/civicase/case/list/directives/sticky-table-header.directive.spec.js]: This test is a bit off. Which element is going to have the `min-width` style set? What will its value be? What would happen if another element gets a `min-width` style?

File: ang/test/civicase/case/list/directives/sticky-table-header.directive.spec.js [hunk 2]
```diff
@@ -104,27 +104,55 @@ describe('civicaseStickyTableHeader directive', function () {
...
+    describe('and viewing the case', function () {
+      beforeEach(inject(function ($timeout) {
+        scope.isLoading = false;
+        scope.caseIsFocused = false;
+        scope.viewingCase = true;
+        scope.$digest();
+      }));
+
+      it('does not add min-width style to the table elements', function () {
+        expect(element.find('thead').html()).not.toContain('style="min-width');
```
Comment (reneolivo) [inline ang/test/civicase/case/list/directives/sticky-table-header.directive.spec.js]: same question

File: ang/test/civicase/case/list/directives/sticky-table-header.directive.spec.js (hunk 1 above)
Comment (deb1990) [reply ang/test/civicase/case/list/directives/sticky-table-header.directive.spec.js]: Fixed.

File: ang/test/civicase/case/list/directives/sticky-table-header.directive.spec.js (hunk 2 above)
Comment (deb1990) [reply ang/test/civicase/case/list/directives/sticky-table-header.directive.spec.js]: Fixed.

**Pull request merged**